magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -257,10 +255,12 @@ object ASG {
 
     val matches = groups.filter(hasExactTagRequirements(_, tagRequirements))
     matches match {
+      case Nil if strategy == Strategy.Dangerous =>
+        reporter.info(s"Unable to locate an autoscaling group AND living dangerously - we'll assume you're creating a new stack")
+        None
       case Nil =>
-        reporter.fail(s"No autoscaling group found with tags $tagRequirements")
-      case List(singleGroup) =>
-        singleGroup
+        reporter.fail(s"No autoscaling group found with tags $tagRequirements. Creating a new stack? Initially choose the ${Strategy.Dangerous} strategy.")
+      case List(singleGroup) => Some(singleGroup)
       case groupList =>
```

Comment:
Can this condition now be reached or is it covered by matching `Nil` without qualification above?

Reply:
I think so, at least it was before...

Reply:
Looking at this i've again i've misread the code, I see how this works now. It does seem like this function is untested though it may be cumbersome to do so given it contains an `AutoScalingClient` and the tests against this file are sparse anyway.